### PR TITLE
Add NoteCard component

### PR DESCRIPTION
**Pull request by CarlSchwan**
image

Already used in the login pages and in a few settings

### CODE REVIEW

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Nice! :) Details:
- [ ] I’d probably cut this border, then it looks cleaner? Also would then look more like the callout boxes in Text
- [ ] Similar with the box-shadow – we don’t really use shadow a lot, and it doesn’t make a lot of sense for this container (we don’t use it for the Text callout boxes either)
- [ ] Is the text color accessible / readable on all the possible backgrounds? 0.2 opacity is used, whereas for color-primary-light we use 10%, so we should probably adjust the values here to 0.1 as well so it’s a bit less intense?

**Comment by jancborchardt** [comment]
Oh and also, this seems like it can be aligned with the notification component? Basically same thing: Container and bar on the left, just one is placed in the content, one on the top right with an x button (and possibly other actions like "undo").

**Comment by CarlSchwan** [comment]
> Nice! :) Details:
> 
>     * [ ]  I’d probably cut this border, then it looks cleaner? Also would then look more like the callout boxes in Text
> 
>     * [ ]  Similar with the box-shadow – we don’t really use shadow a lot, and it doesn’t make a lot of sense for this container (we don’t use it for the Text callout boxes either)
> 
>     * [ ]  Is the text color accessible / readable on all the possible backgrounds? 0.2 opacity is used, whereas for color-primary-light we use 10%, so we should probably adjust the values here to 0.1 as well so it’s a bit less intense?

The text color is accessible with 0.20 and I got a contrast ratio of >11. But I experimented a bit with the various transparency level and I kinda like 15% more:

15%

image

10%

**Comment by nickvergessen** [comment]
Can you also make dark mode screenshots? Would like to see how that turns out.

**Comment by jancborchardt** [comment]
Looks good @CarlSchwan! :) I’d go for 10% so it’s in line with our usage of the primary-light (as will the new design for tags and avatars), and also because it looks much nicer than the slightly darker background colors.

**Comment by vinicius73** [comment]
I guess we can make it closer with `text app callouts` in some way.

callouts

**Comment by CarlSchwan** [comment]
I updated the screenshot

File: src/components/NcNoteCard/NcNoteCard.vue [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.notecard {
+	color: var(--color-text-light) !important;
+	background-color: var(--note-background) !important;
+	border-left: 4px solid var(--note-theme);
```
Comment (tcitworld) [inline src/components/NcNoteCard/NcNoteCard.vue:127]:
```suggestion
	border-inline-start: 4px solid var(--note-theme);
```
If you want to support rtl languages.

**Review by jancborchardt — APPROVED** [review]
Looks great! :) And now this we can also use for the login screen for the "Wrong account name or password" notice

File: src/components/NcNoteCard/NcNoteCard.vue [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+export default {
+	name: 'NcNoteCard',
+
+	props: {
+		/**
+		 * Type of the attribute
+		 */
+		type: {
+			type: String,
+			default: () => 'warning',
```
Comment (raimund-schluessler) [inline src/components/NcNoteCard/NcNoteCard.vue:79]: Doesn't need to be a function
```suggestion
			default: 'warning',
```

File: src/components/NcNoteCard/NcNoteCard.vue [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+			validator: type => type === 'success' || type === 'warning' || type === 'error',
+		},
+		showAlert: {
+			type: Boolean,
+			default: () => false,
```
Comment (raimund-schluessler) [inline src/components/NcNoteCard/NcNoteCard.vue:84]:
```suggestion
			default: false,
```

File: src/components/NcNoteCard/NcNoteCard.vue [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		},
+		heading: {
+			type: String,
+			required: false,
```
Comment (raimund-schluessler) [inline src/components/NcNoteCard/NcNoteCard.vue:88]:
```suggestion
			default: '',
```

File: src/components/NcNoteCard/NcNoteCard.vue [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+	name: 'NcNoteCard',
+
+	props: {
+		/**
+		 * Type of the attribute
+		 */
+		type: {
+			type: String,
+			default: () => 'warning',
+			validator: type => type === 'success' || type === 'warning' || type === 'error',
```
Comment (raimund-schluessler) [inline src/components/NcNoteCard/NcNoteCard.vue:80]:
```suggestion
			validator: type => ['success', 'warning', 'error'].includes(type),
```

**Review by raimund-schluessler — APPROVED** [review]

**Pull request merged**